lib/logstash/outputs/gelf.rb

```diff
@@ -81,12 +81,22 @@ class LogStash::Outputs::Gelf < LogStash::Outputs::Base
   config :short_message, :validate => :string, :default => "short_message"
 
   public
+
+  def inject_client(gelf)
+    @gelf = gelf
+    self
+  end
+
+  def gelf
+    @gelf
+  end
+
   def register
     require "gelf" # rubygem 'gelf'
     option_hash = Hash.new
 
     #@gelf = GELF::Notifier.new(@host, @port, @chunksize, option_hash)
-    @gelf = GELF::Notifier.new(@host, @port, @chunksize)
+    @gelf ||= GELF::Notifier.new(@host, @port, @chunksize)
```

Comment:
Well the idea of having an inject client is more clean in my perspective, it lets you do the same thing (dependency injection) been explicit.